File: Go/Go_1000-/1413.go
// Go
// leetcode 1413
// https://leetcode.cn/problems/minimum-value-to-get-positive-step-by-step-sum/

func minStartValue(nums []int) int {
	res, cur := 1000, 0
	for _, x := range nums {
		cur += x
		if cur < res {
			res = cur
		}
	}
	if res > 0 {
		res = 1
	} else {
		res = 1 - res
	}

	return res
}

File: cpp/cpp_1000-/1413.cc
// C++
// leetcode 1413
// https://leetcode.cn/problems/minimum-value-to-get-positive-step-by-step-sum/

class Solution {
public:
    int minStartValue(vector<int>& nums) {
    	int res = INT_MAX, cur = 0;
    	for (auto x : nums) {
    		cur += x;
    		res = min(res, cur);
    	}
    	if (res > 0)
    		res = 1;
    	else
    		res = 1 - res;
    	return res;
    }
};